aws/aws.go

```diff
@@ -534,7 +534,7 @@ func helperWithTimeoutAndBackoff(metadata *ec2metadata.EC2Metadata, p string) (s
 			result, origErr = metadata.GetUserDataWithContext(ctx)
 		}
 
-		if origErr != nil && (strings.Contains(origErr.Error(), "Client.Timeout exceeded while awaiting headers") || isExponentialError(origErr)) {
+		if origErr != nil && !isErrorCode404(origErr) && (strings.Contains(origErr.Error(), "Client.Timeout exceeded while awaiting headers") || isExponentialError(origErr)) {
```

Comment:
Why isn't isExponentialError() not handling the 404 error signatures? 
Or what was the error which cloudops ended up doing an exponential backoff ?

Reply:
It is a generic Ec2Metadata error with the specific error of 404 that the metadata endpoint doesn't exit.

Reply:
got it